src/main/java/ticTacToe/ui/Ui.java

```diff
@@ -132,4 +140,12 @@ private String promptForNewInput(Grid grid, String mark, Lines lines) {
         promptForPosition(mark, lines, grid);
         return validPosition(grid, mark, lines);
     }
+
+    private String validAnswer(String answer) {
+        if (!answer.equals("y") && !answer.equals("n")) {
```

Comment:
again wrap in a method will help readability - it can be difficult to keep all the negatives in your mind at one time :)